pkg/controllers/infracluster/infracluster_controller.go

```diff
@@ -356,3 +293,37 @@ func getReadiness(infraCluster client.Object) (bool, error) {
 
 	return val, nil
 }
+
+// getRawMAPIProviderSpec returns a raw Machine ProviderSpec from the the cluster.
+func getRawMAPIProviderSpec(ctx context.Context, cl client.Client) ([]byte, error) {
+	var rawProviderSpec []byte
+	if cpms, _ := getActiveCPMS(ctx, cl); cpms != nil {
```

Comment:
This appears to be the only place that we call `getActiveCPMS` and we drop the error. Should we log errors from `Get` and not return them in the function? Or is there a later use that we might have to act on the error?

Reply:
Changed the logic around to check both nil and err